mlflow/models/model.py

```diff
@@ -741,6 +789,50 @@ def log(
                     _logger.warning(_LOG_MODEL_MISSING_INPUT_EXAMPLE_WARNING)
                 elif tracking_uri == "databricks" or get_uri_scheme(tracking_uri) == "databricks":
                     _logger.warning(_LOG_MODEL_MISSING_SIGNATURE_WARNING)
+
+            env_vars = None
+            # validate input example works for serving when logging the model
+            if serving_input and kwargs.get("validate_serving_input", True):
+                from mlflow.models import validate_serving_input
+                from mlflow.utils.model_utils import env_var_tracker
+
+                with env_var_tracker() as env:
+                    try:
+                        validate_serving_input(
+                            model_uri=f"runs:/{run_id}/{mlflow_model.artifact_path}",
+                            serving_input=serving_input,
+                        )
```

Comment:
```suggestion
                        validate_serving_input(
                            model_uri=<local_path>,
                            serving_input=serving_input,
                        )
```

Can we use a local path to avoid downloading the model artifacts?